src/keys.c

```diff
@@ -337,6 +338,12 @@ create_new_keys(const char* jwkdir)
             fprintf(stderr, "Error saving JWK to file (%s)\n", path);
             return 0;
         }
+
+        /* Set 0440 permission for the new key. */
```

Comment:
This way of setting permissions to the keys makes the keys available to read to other users in the system between the time they were written to disk and the change of permissions. There are several exploits and attacks that take advantage of such timings. The answer to that is to do umask(006) at the start of the program or before opening the file.